src/components/Login.js

```diff
@@ -0,0 +1,86 @@
+import React, { useState, useEffect } from "react";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+import { jsPsych } from "jspsych-react";
+import "../App.css";
+import "bootstrap/dist/css/bootstrap.css";
+import "@fortawesome/fontawesome-free/css/all.css";
+
+function Login({ onLogin, envParticipantId, envStudyId, validationFunction }) {
+  // State variables for login screen
+  const dateTimestamp = Date.now();
+  const curDate = new Date(dateTimestamp);
+  const [startDate] = useState(curDate.toString());
+  const [timestamp] = useState(dateTimestamp.toString());
+  const [participantId, setParticipant] = useState("");
+  const [studyId, setStudy] = useState("");
+
+  useEffect(() => {
+    // Update based on environment variables
+    setParticipant(envParticipantId);
+    setStudy(envStudyId);
+  }, [envParticipantId, envStudyId]);
+
+  // Checks if forms are filled in
+  function validateForm() {
+    return participantId.length > 0 && studyId.length > 0;
+  }
+
+  // Function to log in participant
+  function handleSubmit(e) {
+    e.preventDefault();
+    // Validates fields
+    validationFunction(participantId, studyId, startDate, timestamp)
+    // Logs in depending on result from promise
+    .then((loggedIn) => {
```

Comment:
I still think the jspsych piece of this should move to app.  create a wrapper function around `setLogin`, which takes the study and participant id, then if the login state is true, adds them to jspsych.  This would also be able to be used by the mturk piece of code.

It would look something like (in app.js)
```js
const setLoggedIn = (loggedIn, study_id, participant_id) => {
  if (loggedIn) {
    jsPsych.data.addProperties(...)
  }
  setLogin(loggedIn)
}
```